### PR TITLE
fix: docker env parsing after nextjs 13.4.19 upgrade

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
This PR took way longer than I thought but we finally managed to get a closure on this! (hopefully)
The issues are:
- If we have a `.env`, docker compose reads it for its own (docker) configurations and the `/*` is not understood by it
- Next, the `.env.docker` was redundant hence has been removed for a cleaner and one-place-for-all `docker-compose` itself!
- The environment variables can be changed without rebuilding the image else the other env vars are passed as arguments at buildtime hence would need a rebuild.

One point to note is, if a user wants to run the docker method, they have to get rid (rename/backup) of their `.env` file.

This might also need a new release of the docker image on dockerhub with the build time values to be passed as args since we have changed the dockerfile.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
1. Edit the variables in the docker-compose as you like
2. `docker compose up`
3. Verify the changes

## Checklist
- [ ] Added a screen recording or screenshots to this PR
- [x] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,40 @@
...
+    # Uncomment to enable a dedicated connection pool for Prisma using Prisma Data Proxy
+    # Cold boots will be faster and you'll be able to scale your DB independently of your app.
+    # @see https://www.prisma.io/docs/data-platform/data-proxy/use-data-proxy
+    # PRISMA_GENERATE_DATAPROXY=true
+    PRISMA_GENERATE_DATAPROXY:
+
+    # NextJS Auth
+    # @see: https://next-auth.js.org/configuration/options#nextauth_secret
+    # You can use: `openssl rand -base64 32` to generate one
+    NEXTAUTH_SECRET: MoPAA5lQLzx9J8xRa8JWpIwrB4U7STgVImLzI4Xtjag=
```
Comment (mattinannt) [inline docker-compose.yml:23]: As we are now revamping the approach, should we maybe add a mandatory command to generate a secret first? (like we have in the setup script)

File: docker-compose.yml
Comment (mattinannt) [inline docker-compose.yml:1]: We now have variables twice in the file: as args and environment. I think this is confusing. As currently this file is mainly used to get a setup running from source can we maybe only need the build time args and get rid of the environment and maybe later, when we support everything runtime, find a different approach?

Comment (mattinannt) [reply docker-compose.yml:1]: chatGPT recommends referencing the envionment variables in the args to not declaring a variable twice. Maybe we can set all variables in the environment (a the top of the file easy to see) and there divide in build-time and run-time?

Screenshot 2023-09-18 at 11 12 36

File: apps/web/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,77 @@
...
+
+ARG CRON_SECRET
+ENV CRON_SECRET=$CRON_SECRET
+
 WORKDIR /app
 
 COPY . .
-# Copy .env file because Docker don't follow symlinks
-COPY .env.docker /app/apps/web/.env
+RUN touch /app/apps/web/.env
```
Comment (mattinannt) [inline apps/web/Dockerfile:73]: why is this command needed? Shouldn't it also work with no `.env` file?

File: docker-compose.yml (hunk 1 above)
Comment (ShubhamPalriwala) [reply docker-compose.yml:23]: In our docs right?

Comment (mattinannt) [reply docker-compose.yml:23]: Yes, in the docs :-) 
🙏

File: apps/web/Dockerfile (hunk 2 above)
Comment (ShubhamPalriwala) [reply apps/web/Dockerfile:73]: I tried :/, it doesn't work ;( hence passed this empty env

File: docker-compose.yml
Comment (ShubhamPalriwala) [reply docker-compose.yml:1]: Finally managed a way around for this, done!

File: docker-compose.yml (hunk 1 above)
Comment (ShubhamPalriwala) [reply docker-compose.yml]: Added in docs

**Comment by ShubhamPalriwala** [comment]
Hey @mattinannt, i've addressed all the changes, please take a look 🤞🏼

File: apps/formbricks-com/app/docs/self-hosting/from-source/page.mdx [hunk 3]
```diff
@@ -29,7 +29,9 @@ Ensure `docker` & `docker compose` are installed on your server/system. Both are
 
    </CodeGroup>
 
-2.  **Modify the `.env.docker` file as required by your setup.** <br/> This file comes with a basic setup and Formbricks works without making any changes to the file. To enable email sending functionality you need to configure the SMTP settings. If you configured your email credentials, you can also comment the following lines to enable email verification (`# NEXT_PUBLIC_EMAIL_VERIFICATION_DISABLED=1`) and password reset (`# NEXT_PUBLIC_PASSWORD_RESET_DISABLED=1`)
+2.  **Rename your .env file (if it exists)** <br/> The .env file currently works as per NextJs 13.4.19 and not exactly how docker compose likes hence please rename it to something else
```
[the remark this answers is not in the document]
Comment (mattinannt) [reply apps/formbricks-com/app/docs/self-hosting/from-source/page.mdx:32]: If so, please remove this part of the docs

[the remark this answers is not in the document]
Comment (ShubhamPalriwala) [reply apps/formbricks-com/app/docs/self-hosting/from-source/page.mdx:32]: Tried that but didn't work :/ it is always taking the .env

**Pull request merged**